nautobot/core/forms/forms.py

```diff
@@ -56,6 +59,120 @@ def clean(self):
         self.instance.address = self.cleaned_data.get("address")
 
 
+class BaseInlineGFKFormSet(forms.BaseModelFormSet):
```

Comment:
I was talking with John this afternoon and he brought up the idea of an alternate approach - instead of this "SecretsGroup-like" approach where the contact/team association to an object is done in the object's edit view via an embedded formset, maybe we should consider a "device-interfaces-like" approach where adding a new ContactAssociation is done from the "associated contacts" table on an object detail view, i.e. with "Add a Contact" or "Add a Team Contact" button/dropdown in the detail view or a dedicated tab therein.

Reply:
This is the pattern we are talking about here right? 
![Screen Shot 2024-01-08 at 4 14 33 PM](https://github.com/nautobot/nautobot/assets/46973263/ec89f447-cd79-40d2-ab6a-bed51282e7b9)

Reply:
One thing I can think of that can differentiate the Contact/team feature from this approach is adding an existing contact/team versus creating and then adding a **new** contact/team.

I believe for device it is always adding a **new** interface/console port and etc.

Reply:
Good thought, could be either case ideally I think.

Yeah, what you screenshotted there, plus also:

![image](https://github.com/nautobot/nautobot/assets/5603551/410a1eba-33ac-44ca-bf17-7be979692cba)

where we have the ability to do CRUD operations directly on the related-objects table.